presets/presetManager.py

```diff
@@ -0,0 +1,223 @@
+import os.path
+import logging
+import json
+
+logger = logging.getLogger(__name__)
+
+
+class PresetManager(object):
+    MAX_DEPTH = 2
+
+    def __init__(self, paths, strict=False):
+        self.presets = {}
+        self.strict = strict
+
+        # normalize paramter type string -> array
+        if isinstance(paths, basestring):
+            paths = [paths]
+
+        self._loadPaths(paths)
+
+    def _loadPaths(self, paths, depth=0):
+        if depth > self.MAX_DEPTH:
+            return
+
+        for i in range(len(paths)):
+            path = paths[i]
+
+            # avoid empty string
+            if not path:
+                continue
+
+            # cleanUp path
+            if depth == 0:
+                path = os.path.expanduser(path)  # replace ~
+                path = os.path.expandvars(path)  # replace vars
+                path = os.path.normpath(path)  # replace /../ and so on
```

Comment:
because if you apply those functions on successive iterations they simply will do nothing.